storage/filesystem/reference.go

```diff
@@ -28,6 +29,8 @@ func (r *ReferenceStorage) IterReferences() (storer.ReferenceIter, error) {
 		return nil, err
 	}
 
+	reference.Sort(refs)
```

Comment:
We may need a flag to toggle this behaviour just in case it introduces a regression.

Reply:
I've added a new `NewOrderedReferenceSliceIter` instead to create an ordered reference iter

Reply:
I think we should introduce an `IterOrderedReferences` on the storage interface so we don't change the existing behaviour for users that rely on `IterReferences` already, and the new `NewOrderedReferenceSliceIter` is called there only.

Reply:
I dropped the `IterOrderedReferences` altogether. I don't like the API on the storage interface, [Git sorts the references by refname by default](https://github.com/git/git/blob/master/refs/refs-internal.h#L320) and has the option to not to. We can keep things simple and sort the refs where we need to.